extra_data/sourcedata.py

```diff
@@ -70,6 +72,31 @@ def __getitem__(self, key):
             inc_suspect_trains=self.inc_suspect_trains,
         )
 
+    def _get_first_source_file(self):
+        sample_path = self[next(iter(self.keys()))].source_file_paths[0]
```

Comment:
If there are no trains selected, `.source_file_paths` will be an empty list, in which case this will raise IndexError. This is unlike `sourcedata.files`, which we ensure always keeps at least one file even if you deselect all trains.

Are you happy for these properties to just not work with no data selected? Fixing it feels like a lot of effort for a corner case that might well not be important.

Reply:
I would say yes, but thanks for the hint, I will add a custom exception text that makes it explicit why it happened.

Reply:
Added a custom `ValueError` indicating there are no trains to pull source files from.